pom.xml

```diff
@@ -515,6 +518,59 @@
             <sureFireForks9>true</sureFireForks9>
             <artifact.classifier>java9</artifact.classifier>
          </properties>
+
+         <build>
+            <plugins>
+               <plugin>
+                  <groupId>org.apache.maven.plugins</groupId>
+                  <artifactId>maven-antrun-plugin</artifactId>
+                  <executions>
+                     <execution>
+                        <id>compile-java9</id>
+                        <phase>compile</phase>
+                        <goals>
+                           <goal>run</goal>
+                        </goals>
+                        <configuration>
+                           <tasks>
+                              <mkdir dir="${java9.build.outputDirectory}"/>
+                              <copy todir="${java9.build.outputDirectory}">
+                                 <fileset dir="${project.build.outputDirectory}"/>
+                              </copy>
+                              <javac srcdir="${java9.sourceDirectory}" destdir="${java9.build.outputDirectory}"
+                                     classpath="${project.build.outputDirectory}" includeantruntime="false"/>
```

Comment:
In case this ever pops up, if your `module-info.java` file contains dependencies, this can be fixed by:

- Upgrading the plugin to `org.apache.maven.plugins:maven-antrun-plugin:3.0.0`
- Using something like the below configuration instead

```xml
<configuration>
    <target>
        <property name="cp" refid="maven.compile.classpath"/>
        <mkdir dir="${java9.build.outputDirectory}" />
        <copy todir="${java9.build.outputDirectory}">
           <fileset dir="${project.build.outputDirectory}"/>
        </copy>
        <javac srcdir="${java9.sourceDirectory}"
            destdir="${java9.build.outputDirectory}"
            includeantruntime="false"
            modulepath="${cp}">
        </javac>
    </target>
</configuration>
```

This is just a heads-up. It ultimately didn't work for me, and I dropped the attempt of releasing multi version jar files. Might still work for HikariCP, though.

Reply:
If you have a simple build the following is the best/easiest way to go: https://github.com/apache/maven-compiler-plugin/blob/master/src/it/multirelease-patterns/singleproject-runtime/pom.xml#L102-L108

This didn't work for Lukas because his build is quite complex.